src/frontend/dsl/mod.rs

```diff
@@ -310,6 +310,18 @@ impl<'a, F> StepTypeSetupContext<'a, F> {
     }
 }
 
+impl<'a, F: Eq + PartialEq + Hash + Debug + Clone> StepTypeSetupContext<'a, F> {
+    pub fn auto(&mut self, signal: Queriable<F>, expr: ASTExpr<F>) {
+        self.step_type.auto_signals.insert(signal, expr);
+    }
+
+    pub fn auto_eq(&mut self, signal: Queriable<F>, expr: ASTExpr<F>) {
+        self.auto(signal.clone(), expr.clone());
+
+        self.constr(eq(signal, expr));
+    }
```

Comment:
Could we also add `auto_transition`? Not sure how much harder it would be with rotations.

Reply:
There cannot be auto_transition, because the infer signal cannot depend on signals of next steps. Or perhaps it is possible but it would be more complicated to evaluate, so we should leave it as a posible future feature. It is not trivial, because there can be a chain of `nexts` in the auto signals

Reply:
Understood!